src/mds/Locker.cc

```diff
@@ -2049,6 +2055,10 @@ void Locker::resume_stale_caps(Session *session)
     if (cap->is_stale()) {
       dout(10) << " clearing stale flag on " << *in << dendl;
       cap->clear_stale();
+
+      if (in->state_test(CInode::STATE_EXPORTINGCAPS))
+	continue;
```

Comment:
This is subtle; can you explain in the commit message? I think this is assuming we're the recipient and can just toss out the stale caps now, right?